File: frontend_web/src/utils/cleanFileName.ts
export default function sanitizeFileName(name: string): string {
    return name
        .normalize("NFD") // Normalization Form Decomposition è => e + `
        .replace(/[\u0300-\u036f]/g, "") // enlève les accents
        .replace(/[^a-zA-Z0-9.\-_]/g, "_"); // remplace ce qui n'est pas alphanumérique par un _
}

File: backend_api/src/main/java/com/etudia/etudia/utils/CleanFileName.java
package com.etudia.etudia.utils;

public class CleanFileName {

    public static String clean(String name) {
        // Normalization Form Decomposition è => e + `
        String normalized = java.text.Normalizer.normalize(name, java.text.Normalizer.Form.NFD);
        // enlève les accents
        normalized = normalized.replaceAll("[\\p{InCombiningDiacriticalMarks}]", "");
        // remplace ce qui n'est pas alphanumérique par un _
        return normalized.replaceAll("[^a-zA-Z0-9.\\-_]", "_");
    }

}